vxquery-core/src/main/java/org/apache/vxquery/runtime/functions/index/IndexConstructorUtil.java

```diff
@@ -111,25 +127,41 @@ public static void evaluate(TaggedValuePointable[] args, IPointable result, UTF8
      * it indexes that document node.
      */
     public static void indexXmlFiles(File collectionDirectory, IndexWriter writer, boolean isElementPath,
-            TaggedValuePointable nodep, ArrayBackedValueStorage abvsFileNode, ITreeNodeIdProvider nodeIdProvider,
-            SequenceBuilder sb, ByteBufferInputStream bbis, DataInputStream di, String nodeId)
-                    throws SystemException, IOException {
+                                     TaggedValuePointable nodep, ArrayBackedValueStorage abvsFileNode, ITreeNodeIdProvider nodeIdProvider,
+                                     SequenceBuilder sb, ByteBufferInputStream bbis, DataInputStream di, String nodeId)
+            throws SystemException, IOException {
+
+
         for (File file : collectionDirectory.listFiles()) {
 
             if (readableXmlFile(file.getPath())) {
                 abvsFileNode.reset();
                 // Get the document node
-                XMLParser parser = new XMLParser(false, nodeIdProvider, nodeId);
-                FunctionHelper.readInDocFromString(file.getPath(), bbis, di, abvsFileNode, parser);
-
-                nodep.set(abvsFileNode.getByteArray(), abvsFileNode.getStartOffset(), abvsFileNode.getLength());
-
-                //Add the document to the index
-                //Creates one lucene doc per file
-                IndexDocumentBuilder ibuilder = new IndexDocumentBuilder(nodep, writer);
+//                XMLParser parser = new XMLParser(false, nodeIdProvider, nodeId);
+//                FunctionHelper.readInDocFromString(file.getPath(), bbis, di, abvsFileNode, parser);
+//
+//                nodep.set(abvsFileNode.getByteArray(), abvsFileNode.getStartOffset(), abvsFileNode.getLength());
+//
+//                //Add the document to the index
+//                //Creates one lucene doc per file
+//                IndexDocumentBuilder ibuilder = new IndexDocumentBuilder(nodep, writer, file.getCanonicalPath());
+                IndexDocumentBuilder ibuilder = getIndexBuilder(file, writer, nodep, abvsFileNode, nodeIdProvider,
+                        bbis, di, nodeId);
 
                 ibuilder.printStart();
 
+                if (!isMetaFilePresent) {
+                    XmlMetadata xmlMetadata = new XmlMetadata();
+                    xmlMetadata.setPath(file.getCanonicalPath());
+                    xmlMetadata.setFileName(file.getName());
+                    try {
+                        xmlMetadata.setMd5(metaFileUtil.generateMD5(file));
+                    } catch (NoSuchAlgorithmException e) {
+                        e.printStackTrace();
```

Comment:
Please don't swallow exceptions. One option would be to wrap the exception into a SystemException with error code SYSE0001.